tests/rados/test_noscrub_nodeepscrub_check.py

```diff
@@ -0,0 +1,218 @@
+        method_should_succeed(wait_for_clean_pg_sets, rados_object, timeout=3600)
+        # Deep-scrub should not start immediately
+        init_time, _ = installer.exec_command(
+            cmd="sudo date '+%Y-%m-%dT%H:%M:%S.%3N+0000'"
+        )
+        init_time = init_time.strip()
+        exec_end_time = datetime.datetime.now() + datetime.timedelta(seconds=20)
+        while exec_end_time > datetime.datetime.now():
+            end_time, _ = installer.exec_command(
+                cmd="sudo date '+%Y-%m-%dT%H:%M:%S.%3N+0000'"
+            )
+            end_time = end_time.strip()
+            for pg_id in pg_id_list:
+                log_lines = [
+                    rf"{pg_id}\s*(?!deep-)scrub starts",
+                    rf"{pg_id}\s*deep-scrub starts",
+                ]
+                if verify_scrub_log(
+                    rados_object, init_time, end_time, pg_id, log_lines
+                ):
+                    log.error(
+                        f"The scrub started on the {pg_id} after setting the no_scrub and no_deepscrub flag"
+                    )
+                    return 1
+            time.sleep(30)
+    except Exception as e:
+        log.info(e)
+        log.info(traceback.format_exc())
+        return 1
+    finally:
+        log.info("============Execution of finally block==================")
+        mon_object.remove_config(section="osd", name="debug_osd")
+        method_should_succeed(rados_object.delete_pool, pool_name)
+        scrub_obj.set_osd_flags("unset", "nodeep-scrub")
+        scrub_obj.set_osd_flags("unset", "noscrub")
+        mon_object.remove_config(section="osd", name="osd_scrub_min_interval")
+        mon_object.remove_config(section="osd", name="osd_deep_scrub_interval")
+        time.sleep(10)
+        # log cluster health
+        rados_object.log_cluster_health()
+        # check for crashes after test execution
+        if rados_object.check_crash_status():
+            log.error("Test failed due to crash at the end of test")
+            return 1
+    return 0
+
+
+def verify_scrub_log(rados_object, init_time, end_time, pgid, log_lines):
+    """
+    Method to check the scrub and deep-scrub logs exists or not
+    Args:
+        rados_object: Rados object
+        init_time:  initial time
+        end_time:  End time
+        pgid: pg id
+        log_lines: log lines to check in th logs
+
+    Returns: True -> If log lines are present in the logs
+             False ->  Log lines not exists in the logs
+    """
+    log.info(f"The log lies to check - {log_lines}")
+    acting_set = rados_object.get_pg_acting_set(pg_num=pgid)
+    log.info(f"The acting osd set of the pg {pgid} is {acting_set}")
+    fsid = rados_object.run_ceph_command(cmd="ceph fsid")["fsid"]
+    scrub_flag = False
+    for osd_id in acting_set:
+        host = rados_object.fetch_host_node(daemon_type="osd", daemon_id=osd_id)
+        cmd_get_log_lines = (
```

Comment:
We need to make sure that the log_to_file is enabled on the cluster for the log files to be present in the location. Please add a check for the same at the beginning of the test scenario.

Reply:
Added the code at the beginning of the code.